From c79bcda0c1a62a8b1000166126c37b0be08ff65c Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 2 Jan 2025 11:11:46 +0000
Subject: [PATCH] convert: Mention "operating system" in error message

A member of Red Hat's support team mentioned that the error "source
guest" is confusing.  Since this refers to the operating system inside
the source guest, say that.

(cherry picked from commit f58b02a64f4238d7e2ddec45082a6f9b2c98a0c8)
---
 convert/inspect_source.ml | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/convert/inspect_source.ml b/convert/inspect_source.ml
index 78447193..218e517a 100644
--- a/convert/inspect_source.ml
+++ b/convert/inspect_source.ml
@@ -141,11 +141,11 @@ let rec inspect_source root_choice g =
 and choose_root root_choice g = function
   | [] ->
      error (f_"inspection could not detect the source guest \
-               (or physical machine).\n\nAssuming that you are \
-               running virt-v2v/virt-p2v on a source which is \
-               supported (and not, for example, a blank disk), \
-               then this should not happen.\n\nNo root device \
-               found in this operating system image.");
+               (or physical machine) operating system.\n\n\
+               Assuming that you are running virt-v2v/virt-p2v \
+               on a source which is supported (and not, for example, \
+               a blank disk), then this should not happen.\n\n\
+               No root device found in this operating system image.");
   | [root] -> root (* only one root, so return it *)
   | roots ->
      (* If there are multiple roots, use the [--root] option supplied
@@ -254,8 +254,9 @@ and sanity_check_inspection inspect =
 and error_if_unknown fieldname value =
   if value = "unknown" then
     error (f_"inspection could not detect the source guest (or \
-              physical machine).\n\nAssuming that you are running \
-              virt-v2v/virt-p2v on a source which is supported (and \
-              not, for example, a blank disk), then this should not \
-              happen.\n\nInspection field ‘%s’ was ‘unknown’.")
+              physical machine) operating system.\n\n\
+              Assuming that you are running virt-v2v/virt-p2v \
+              on a source which is supported (and not, for example, \
+              a blank disk), then this should not happen.\n\n\
+              Inspection field ‘%s’ was ‘unknown’.")
           fieldname
